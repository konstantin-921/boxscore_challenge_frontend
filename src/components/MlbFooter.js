import React from 'react'
import { withStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'

const styles = theme => ({
  container: {
    textAlign: 'center',
    height: '80px', 
    fontSize: '30px',
    fontWeight: 'bold',
    color: '#ffffff'
  }
});

class MlbFooter extends React.Component {

  render() {
    const { classes, data } = this.props
    const teamAway = data.away_team.last_name
    const teamHome = data.home_team.last_name
    const hitsAway = data.away_batter_totals.hits
    const batsAway = data.away_batter_totals.at_bats
    const hitsHome = data.home_batter_totals.hits
    const batsHome = data.home_batter_totals.at_bats
    // const status = data.event_information.
    console.log('====================================');
    console.log(this.props.data);
    console.log('====================================');
    return (
      <div style={{ marginBottom: '100px', width: '100%', minWidth: '410px', }}>
        <Grid container spacing={0} direction="row" alignItems="center" justify="center">
            <Grid item xs={5}>
              <div className={classes.container} style={{backgroundColor: 'blue'}}>
                <div style={{ paddingTop: 10 }}>{teamAway}</div>
                <div style={{ fontSize: '15px' }}>{hitsAway}-{batsAway}</div>
              </div>
              
            </Grid>
            <Grid item xs={2}>
              <div className={classes.container} style={{ backgroundColor: 'white', borderBottom: '0.5px solid #e0e0e0' }}>

              </div>
            </Grid>
            <Grid item xs={5}>
              <div className={classes.container} style={{ backgroundColor: 'red' }}>
                <div style={{ paddingTop: 10 }}>{teamHome}</div>
                <div style={{ fontSize: '15px' }}>{hitsHome}-{batsHome}</div>
              </div>
            </Grid>
        </Grid>
      </div>
    );
  }
}

export default withStyles(styles)(MlbFooter);