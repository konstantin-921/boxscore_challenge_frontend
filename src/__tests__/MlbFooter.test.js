import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import MlbFooter from '../components/MlbFooter'

describe('MlbFooter component', () => {

  const props = {
    away_team: { last_name: 'CUBS' },
    home_team: { last_name: 'CARDINALS' },
    away_batter_totals: { hits: 7, at_bats: 30 },
    home_batter_totals: { hits: 5, at_bats: 25 },
    event_information: { status: 'completed' }
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(<MlbFooter data={props} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

})