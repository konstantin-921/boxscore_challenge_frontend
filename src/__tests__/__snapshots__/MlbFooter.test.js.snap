// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MlbFooter component renders correctly 1`] = `
<div
  style={
    Object {
      "marginBottom": "100px",
    }
  }
>
  <div
    className="MuiGrid-container-3 MuiGrid-align-items-xs-center-11 MuiGrid-justify-xs-center-20"
  >
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-5-36"
    >
      <div
        className="MlbFooter-container-1"
        style={
          Object {
            "backgroundColor": "blue",
          }
        }
      >
        <div
          style={
            Object {
              "paddingTop": 10,
            }
          }
        >
          CUBS
        </div>
        <div
          style={
            Object {
              "fontSize": "15px",
            }
          }
        >
          7
          -
          30
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-2-33"
    >
      <div
        className="MlbFooter-middleContainer-2"
      >
        <div>
          BTM
        </div>
        <div>
          9TH
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-5-36"
    >
      <div
        className="MlbFooter-container-1"
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      >
        <div
          style={
            Object {
              "paddingTop": 10,
            }
          }
        >
          CARDINALS
        </div>
        <div
          style={
            Object {
              "fontSize": "15px",
            }
          }
        >
          5
          -
          25
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MlbSideTable component renders correctly 1`] = `
<div
  style={
    Object {
      "marginBottom": "100px",
    }
  }
>
  <div
    className="MuiGrid-container-3 MuiGrid-align-items-xs-center-11 MuiGrid-justify-xs-center-20"
  >
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-5-36"
    >
      <div
        className="MlbFooter-container-1"
        style={
          Object {
            "backgroundColor": "blue",
          }
        }
      >
        <div
          style={
            Object {
              "paddingTop": 10,
            }
          }
        >
          CUBS
        </div>
        <div
          style={
            Object {
              "fontSize": "15px",
            }
          }
        >
          7
          -
          30
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-2-33"
    >
      <div
        className="MlbFooter-middleContainer-2"
      >
        <div>
          BTM
        </div>
        <div>
          9TH
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-5-36"
    >
      <div
        className="MlbFooter-container-1"
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      >
        <div
          style={
            Object {
              "paddingTop": 10,
            }
          }
        >
          CARDINALS
        </div>
        <div
          style={
            Object {
              "fontSize": "15px",
            }
          }
        >
          5
          -
          25
        </div>
      </div>
    </div>
  </div>
</div>
`;
